Libraries/DirectMLX.h

```diff
@@ -3470,9 +3475,48 @@ namespace dml
         return out;
     }
 
-    // 
-    // TODO: NonZeroCoordinates
-    // 
+    struct NonZeroCoordinatesOutputs
+    {
+        Expression count;
+        Expression coordinates;
+    };
+    inline NonZeroCoordinatesOutputs NonZeroCoordinates(Expression input)
+    {
+        detail::GraphBuilder* builder = input.Impl()->GetGraphBuilder();
+        TensorDesc inputTensor = input.Impl()->GetOutputDesc();
+        const auto& inputTensorSizes = inputTensor.sizes;
+        uint32_t dimensionCount = static_cast<uint32_t>(inputTensorSizes.size());
+
+        TensorDimensions outputSizes = {};
+        uint32_t effectiveRank = -1;
+        uint32_t totalElements = 1;
+        for (uint32_t i = 0; i < dimensionCount; ++i)
+        {
+            if (inputTensorSizes[i] > 1 && effectiveRank == -1)
+            {
+                effectiveRank = dimensionCount - i;
+            }
+            totalElements *= inputTensorSizes[i];
+            outputSizes.push_back(1);
```

Comment:
If `dimensionCount == 1`, wouldn't `outputSizes` only have 1 dimension? We might need to pad `outputSizes` in that case, otherwise you will have an AV below when you try to insert `totalElements`.

Reply:
I was proposing to Ryan that he just always passes:

`TensorDimensions outputSizes = {totalElements, inputDimensionCount};`

It eliminates all 3 of these cases, the assert, padding issue, and overflow when all are 1 (and it makes the code simpler).

Reply:
Yeah I am just going to use outputSizes = {totalElements, inputDimensionCount}